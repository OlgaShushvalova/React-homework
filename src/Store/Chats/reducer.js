import { ADD_CHAT_ACTION, DELETE_CHAT_ACTION } from "./constants";
import { v4 as uuidv4 } from "uuid";

const initialState = {
  chatList: [],
};

export const chatsReducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_CHAT_ACTION: {
      return {
        ...state,
        chatList: [
          ...state.chatList,
          { id: uuidv4(), name: action.payload.name },
        ],
      };
    }
    case DELETE_CHAT_ACTION: {
      const filteredChats = state.chatList.filter(
        ({ id }) => id !== action.payload.chatId
      );
      return {
        ...state,
        chatList: [...filteredChats],
      };
    }
    default:
      return state;
  }
};
