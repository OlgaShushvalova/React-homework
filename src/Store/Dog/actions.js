import { DOG_URL } from "../../Api";
import { GET_DOG_REQUEST, GET_DOG_SUCCESS, GET_DOG_ERROR } from "./constants";

export const getDogRequestAction = () => ({
  type: GET_DOG_REQUEST,
});

export const getDogSuccessAction = (dogUrl) => ({
  type: GET_DOG_SUCCESS,
  payload: dogUrl,
});

export const getDogErrorAction = () => ({
  type: GET_DOG_ERROR,
});

export const getDogActionWithThunk = () => async (dispatch, getState) => {
  dispatch(getDogRequestAction());

  try {
    const response = await fetch(DOG_URL);

    if (!response.ok) {
      throw new Error("error getting data whith status");
    }
    const result = await response.json();

    dispatch(getDogSuccessAction(result.message));
  } catch (error) {
    dispatch(getDogErrorAction());
  }
};
