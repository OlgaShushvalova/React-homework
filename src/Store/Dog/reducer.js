import { GET_DOG_REQUEST, GET_DOG_SUCCESS, GET_DOG_ERROR } from "./constants";

const initialState = {
  dogUrl: undefined,
  loading: true,
  error: false,
};

export const dogReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_DOG_REQUEST: {
      return {
        ...state,
        loading: true,
        error: false,
      };
    }
    case GET_DOG_SUCCESS: {
      return {
        ...state,
        loading: false,
        dogUrl: action.payload,
      };
    }
    case GET_DOG_ERROR: {
      return {
        ...state,
        loading: false,
        error: true,
      };
    }
    default:
      return state;
  }
};
