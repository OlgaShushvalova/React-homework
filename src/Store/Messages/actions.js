import { ADD_MESSAGE_ACTION } from "./constants";

export const addMessageAction = ({ chatId, author, message }) => ({
  type: ADD_MESSAGE_ACTION,
  payload: { chatId, author, message },
});
