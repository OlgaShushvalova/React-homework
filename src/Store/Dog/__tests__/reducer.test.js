import { getDogRequestAction } from "../actions";
import { dogReducer } from "../reducer";

describe("test dog reducer", () => {
  it("test getDogRequestAction", () => {
    const received = dogReducer(undefined, getDogRequestAction());

    expect(received).toMatchSnapshot();
  });
});
