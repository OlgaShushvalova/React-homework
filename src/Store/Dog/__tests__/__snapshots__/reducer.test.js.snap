// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test dog reducer test getDogRequestAction 1`] = `
Object {
  "dogUrl": undefined,
  "error": false,
  "loading": true,
}
`;
